models/aave/optimism/aave_v3_optimism_borrow.sql

```diff
@@ -0,0 +1,81 @@
+{{ config(
+    schema = 'aave_v3_optimism'
+    , alias='borrow'
+    , post_hook='{{ expose_spells(\'["optimism"]\',
+                                  "project",
+                                  "aave_v3",
+                                  \'["batwayne", "chuxinh"]\') }}'
+  )
+}}
+
+SELECT
+      version,
+      transaction_type,
+      loan_type,
+      erc20.symbol,
+      borrow.token as token_address,
+      borrower,
+      repayer,
+      liquidator,
+      amount / concat('1e',erc20.decimals) AS amount,
+      (amount/ concat('1e',p.decimals)) * price AS usd_amount,
+      evt_tx_hash,
+      evt_index,
+      evt_block_time,
+      evt_block_number   
+FROM (
+SELECT 
+    '3' AS version,
+    'borrow' AS transaction_type,
+    CASE 
+        WHEN interestRateMode = '1' THEN 'stable'
+        WHEN interestRateMode = '2' THEN 'variable'
+    END AS loan_type,
+    reserve AS token,
+    user AS borrower, 
+    NULL::string AS repayer,
+    NULL::string AS liquidator,
+    amount, 
+    evt_tx_hash,
+    evt_index,
+    evt_block_time,
+    evt_block_number
+FROM {{ source('aave_v3_optimism','Pool_evt_Borrow') }} 
+UNION ALL 
+SELECT 
+    '3' AS version,
+    'repay' AS transaction_type,
+    NULL AS loan_type,
+    reserve AS token,
+    user AS borrower,
+    repayer AS repayer,
+    NULL::string AS liquidator,
+    - amount AS amount,
+    evt_tx_hash,
+    evt_index,
+    evt_block_time,
+    evt_block_number
+FROM {{ source('aave_v3_optimism','Pool_evt_Repay') }}
+UNION ALL
+SELECT 
+    '3' AS version,
+    'borrow_liquidation' AS transaction_type,
+    NULL AS loan_type,
+    debtAsset AS token,
+    user AS borrower,
+    liquidator AS repayer,
+    liquidator AS liquidator,
+    - debtToCover AS amount,
+    evt_tx_hash,
+    evt_index,
+    evt_block_time,
+    evt_block_number
+FROM {{ source('aave_v3_optimism','Pool_evt_LiquidationCall') }}
+) borrow
+LEFT JOIN {{ ref('tokens_optimism_erc20') }} erc20
+    ON borrow.token = erc20.contract_address
+LEFT JOIN {{ source('prices','usd') }} p 

```

Comment:
I think there are cases when the erc20 token address on optimism is not the same as that on ETH and also missing in `prices.usd` here. Is that the case for tokens available to borrow on Aave?

Reply:
Yes, for some tokens the price is missing at certain periods.

Reply:
I'm thinking we either have to fill those missing `usd` values or leave them out for now because it's not reflecting them correctly

Reply:
I have swapped the tokens from OP to ETH for the price data in order to fill the missing prices.